tools/wptrunner/wptrunner/update/tree.py

```diff
@@ -183,6 +189,20 @@ def add_new(self, prefix=None):
             args = ("--no-ignore-removal", prefix)
         self.git("add", *args)
 
+    def add_ignored(self, prefix=None):
+        """Add files to the staging area that are explicitly ignored by git.
+
+        :param prefix: None to include all files or a path prefix to
+                       add all files under that path.
+        """
+        ignored_files = self.git("status", "--short", "--ignored", prefix)
```

Comment:
I don't think this does what you want since it includes files that actually ought to be ignored. So far I only found `git ls-tree -r --name-only HEAD | git check-ignore --no-index --stdin` to list files that would be ignored but aren't.

Reply:
I'm not exactly clear on what "actually ought to be ignored" means. The command in this PR only returns .coveragerc; the command you suggest returns that as well as a couple testharness_runner.html files.

Reply:
Mmm, when I run the status-based command in my web-platform-tests clone, I get a bunch of pyc files. In the sync tree those don't exist, though.

Reply:
Yeah, it's the latter I'm worried about because if you start causing those to exist (e.g. by running the manifest update at a different point during sync or similar) then you end up accidentially checking them in. I think the files you want is the intersection of files in the working tree that match a gitignore rule and files that are in the HEAD commit.